File: rest/src/main/java/io/sparkled/viewmodel/stage/StageViewModel.java
package io.sparkled.viewmodel.stage;

import io.sparkled.viewmodel.ViewModel;
import io.sparkled.viewmodel.stage.prop.StagePropViewModel;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class StageViewModel implements ViewModel {
    private Integer id;
    private String name;
    private Integer width;
    private Integer height;
    private List<StagePropViewModel> stageProps = new ArrayList<>();

    public Integer getId() {
        return id;
    }

    public StageViewModel setId(Integer id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public StageViewModel setName(String name) {
        this.name = name;
        return this;
    }

    public Integer getWidth() {
        return width;
    }

    public StageViewModel setWidth(Integer width) {
        this.width = width;
        return this;
    }

    public Integer getHeight() {
        return height;
    }

    public StageViewModel setHeight(Integer height) {
        this.height = height;
        return this;
    }

    public List<StagePropViewModel> getStageProps() {
        return stageProps;
    }

    public StageViewModel setStageProps(List<StagePropViewModel> stageProps) {
        this.stageProps = stageProps;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        StageViewModel that = (StageViewModel) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(name, that.name) &&
                Objects.equals(width, that.width) &&
                Objects.equals(height, that.height) &&
                Objects.equals(stageProps, that.stageProps);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, width, height, stageProps);
    }

    @Override
    public String toString() {
        return "StageViewModel{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", width=" + width +
                ", height=" + height +
                ", stageProps=" + stageProps +
                '}';
    }
}

File: rest/src/main/kotlin/io/sparkled/viewmodel/stage/StageViewModel.kt
package io.sparkled.viewmodel.stage

import io.sparkled.viewmodel.ViewModel
import io.sparkled.viewmodel.stage.prop.StagePropViewModel

class StageViewModel : ViewModel {
    private var id: Int? = null
    private var name: String? = null
    private var width: Int? = null
    private var height: Int? = null
    private var stageProps: List<StagePropViewModel> = emptyList()

    fun getId(): Int? {
        return id
    }

    fun setId(id: Int?): StageViewModel {
        this.id = id
        return this
    }

    fun getName(): String? {
        return name
    }

    fun setName(name: String?): StageViewModel {
        this.name = name
        return this
    }

    fun getWidth(): Int? {
        return width
    }

    fun setWidth(width: Int?): StageViewModel {
        this.width = width
        return this
    }

    fun getHeight(): Int? {
        return height
    }

    fun setHeight(height: Int?): StageViewModel {
        this.height = height
        return this
    }

    fun getStageProps(): List<StagePropViewModel> {
        return stageProps
    }

    fun setStageProps(stageProps: List<StagePropViewModel>): StageViewModel {
        this.stageProps = stageProps
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as StageViewModel

        if (id != other.id) return false
        if (name != other.name) return false
        if (width != other.width) return false
        if (height != other.height) return false
        if (stageProps != other.stageProps) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id ?: 0
        result = 31 * result + (name?.hashCode() ?: 0)
        result = 31 * result + (width ?: 0)
        result = 31 * result + (height ?: 0)
        result = 31 * result + stageProps.hashCode()
        return result
    }

    override fun toString(): String {
        return "StageViewModel(id=$id, name=$name, width=$width, height=$height, stageProps=$stageProps)"
    }
}
